test/unit/modules/blocks/chain.js

```diff
@@ -15,313 +15,596 @@
+				it('should call a callback with error', done => {
+					blocksChainModule.deleteAfterBlock(1, err => {
+						expect(err).to.equal('Blocks#deleteAfterBlock error');
+						expect(loggerStub.error.args[0][0]).to.contains(
+							'deleteAfterBlock-ERR'
 						);
+						done();
+					});
+				});
+			});
 
-						describe('when modules.accounts.getAccount fails', () => {
-							it('should call library.logger.error with error');
-
-							it(
-								'should call library.logger.error "Failed to undo transactions"'
-							);
-
-							it('should call process.exit with 0');
-						});
-
-						describe('when modules.accounts.getAccount succeeds', () => {
-							it('should call modules.transactions.undo');
+			describe('when succeeds', () => {
+				beforeEach(() => {
+					return library.db.blocks.deleteAfterBlock.resolves(true);
+				});
 
-							it('should call modules.transactions.undo with transaction');
+				it('should call a callback with no error and res data', done => {
+					blocksChainModule.deleteAfterBlock(1, (err, res) => {
+						expect(err).to.be.null;
+						expect(res).to.be.true;
+						done();
+					});
+				});
+			});
+		});
+	});
 
-							it('should call modules.transactions.undo with oldLastBlock');
+	describe('applyGenesisBlock', () => {
+		let applyTransactionTemp;
 
-							it('should call modules.transactions.undo with sender');
+		beforeEach(done => {
+			modules.rounds.tick.callsArgWith(1, null, true);
+			applyTransactionTemp = __private.applyTransaction;
+			__private.applyTransaction = sinonSandbox.stub();
+			done();
+		});
 
-							it('should call modules.transactions.undoUnconfirmed');
+		afterEach(done => {
+			__private.applyTransaction = applyTransactionTemp;
+			done();
+		});
 
-							it(
-								'should call modules.transactions.undoUnconfirmed with transaction'
-							);
+		describe('when block.transactions is empty', () => {
+			it('modules.rouds.tick should call a callback', done => {
+				blocksChainModule.applyGenesisBlock(blockWithEmptyTransactions, () => {
+					expect(modules.blocks.utils.getBlockProgressLogger.calledOnce).to.be
+						.true;
+					expect(modules.blocks.lastBlock.set.calledOnce).to.be.true;
+					expect(modules.blocks.lastBlock.set.args[0][0]).to.deep.equal(
+						blockWithEmptyTransactions
+					);
+					expect(modules.rounds.tick.args[0][0]).to.deep.equal(
+						blockWithEmptyTransactions
+					);
+					done();
+				});
+			});
+		});
 
-							it('should call self.deleteBlock');
+		describe('when block.transactions is not empty', () => {
+			describe('modules.accounts.setAccountAndGet', () => {
+				describe('when fails', () => {
+					beforeEach(() => {
+						return modules.accounts.setAccountAndGet.callsArgWith(
+							1,
+							'setAccountAndGet-ERR',
+							true
+						);
+					});
 
-							it('should call self.deleteBlock with oldLastBlock.id');
+					it('should return process.exit(0)', done => {
```

Comment:
Please change it from `should return process.exit(0)`
to `should call process.exit with 0`
Also, you are not checking it in your tests. Please apply it here and in the following `process.exit(0)` tests.